include/qore/intern/BufferedStreamReader.h

```diff
@@ -90,24 +109,69 @@ class BufferedStreamReader : public StreamReader {
          // Won't overflow because the buffer actually has real capacity of bufCapacity+1.
          buf[bufCount] = '\0';
 
+         //printd(5, "BufferedStreamReader::readLine() buf: '%s' bufCount: " QLLD "\n", buf, bufCount);
+
          qore_size_t eolLen;
-         const char* p = findEolInBuffer(*eolstr, eolLen, rc==0);
+         char this_pmatch;
+         const char* p = findEolInBuffer(*eolstr, eolLen, rc==0, this_pmatch);
          assert(eolLen >= 0);
 
          if (p) { // Found end of line.
             assert(p >= buf);
             qore_size_t dataSize = p - buf;
             line->concat(buf, dataSize + (trim ? 0 : eolLen));
-            // Move remaining data from middle to the front of the buffer.
+            assert(bufCount >= (dataSize + eolLen));
+            // Move remaining data to the front of the buffer.
             bufCount -= dataSize + eolLen;
-            memmove(buf, p + eolLen, bufCount);
+            if (bufCount)
+               memmove(buf, p + eolLen, bufCount);
             return line.release();
          }
          else {
             if (rc > 0) {
-               qore_size_t dataSize = (bufCount >= eolLen) ? bufCount - eolLen : 0;
+               if (eol) {
+                  // we read in new data, so here we need to look for a partial eol match at the end of the buffer
```

Comment:
@davidnich Why did you add so much new code here? The previous code worked just fine, including correctly handling EOL divided by the edge of the buffer.